src/lua/funcs/lua_registry.rs

```diff
@@ -0,0 +1,131 @@
+//! Metamethods for accesssing the registry values!
+
+use super::super::super::registry;
+use registry::RegistryAccess;
+
+use hlua::{LuaTable};
+use hlua::any::AnyLuaValue;
+
+use rustc_serialize::json::{Json, ToJson};
+use std::ops::Deref;
+pub fn index(table: AnyLuaValue, lua_key: AnyLuaValue)
+             -> Result<AnyLuaValue, &'static str> {
+    if let AnyLuaValue::LuaString(key) = lua_key {
```

Comment:
I don't think I've tested thoroughly what happens with invalid types on hlua calling Rust functions.
I think I set it up this way because it's the metatable reference, and a metatable could be indexed with anything (similar issues arise with Lua -> Json conversion). 

The upcoming `async-lua